### ISSUE TITLE
Question about dependencies - gopkg.in/macaroon.v2-unstable and gopkg.in/macaroon-bakery.v2-unstable

### ISSUE THREAD

**Issue by nodens** [lxc/lxd#4243]
Hi,

I'm working on packaging LXD in Debian.

I would like to know if there is a reason why you decided to import the v2-unstable branch for macaroon-related stuff ?

Wouldn't the v2 make more sense ? The unstable version hasn't been updated as recently as the v2 (but maybe that's the reason) ?

I see that the corresponding code has been removed from the ubuntu package, probably to avoid adding dependencies to a LTS release.

macaroon.v2 and macaroon-bakery.v2 is available in Debian, I'm going to use that for packaging, should I expect any issue ?

Anyway, thanks for working on LXD :)

Cheers,

**Comment by pcdummy** [comment]
GREAT, LXD in Debian soon. Thank you!

**Comment by stgraber** [comment]
I don't believe there's a reason for that, other than the v2 branch didn't exist back then as v2 wasn't stable yet.

What LXD release are you packaging for Debian? We usually recommend distros with a long support cycle such as Debian to stick to our LTS branch (2.0.11 right now). If that's what you're packaging, then you may be interested by https://github.com/lxc/lxd-pkg-ubuntu/tree/dpm-xenial/debian/patches which are the patches we use on top of the 2.0.11 release tarball in Ubuntu, one of those patches removes the dependency on macaroon/bakery entirely as that's not actually used in 2.0.11.

**Comment by stgraber** [comment]
Ah, actually, current LXD master uses the regular v2 branches rather than v2-unstable, so we did the switch already (back in later October apparently).

**Comment by nodens** [comment]
Thanks for your answers! And please, tell me if I should use another communication medium than GH issues for such questions (maybe discuss.linuxcontainers.org ?).

Yes, that's the 2.0.11 I'm working on. I saw the patches indeed in the ubuntu package, I intend to use then when that makes sense.

The thing is, in the 2.0.11 release it's still the -unstable part imported, and since the packages are available in Debian there is no reason to remove this code if it's working. 

Do you plan to release a 2.0.12 soon ? If not I'll patch it in the debian package until it's out.

Cheers,

**Comment by stgraber** [comment]
We're currently rushing for the 3.0 release so I don't expect to tag 2.0.12 until after 3.0 is out (or possibly at the same time). We're talking a few weeks here but I don't really want to delay you packaging 2.0 in Debian for that as it's long overdue :)

The 2.0.x branch is never supposed to add new dependencies so it pulling bakery and macaroon was a mistake that happened as the result of some api cherry-picks in the client codebase. I noticed it when packaging in Ubuntu (which has the same Go requirements as Debian), fixed it in the upstream stable branch and then cherry-picked that fix into the Ubuntu package, eliminating that particular dependency and getting us back to the same dependency list as 2.0.10 and all releases before.

For such discussions, the forum may be a better place unless there's an actual bug to be fixed. I guess in the case of this issue it seemed like there was, except it already had been fixed :)
